test/server.test.ts

```diff
@@ -0,0 +1,30 @@
+import { describe, beforeEach, it } from 'mocha';
+import * as expect from 'expect';
+
+import { Response } from 'node-fetch';
+
+import { main } from '../src/main';
+import { actions } from './helpers';
+
+describe("server", () => {
+  beforeEach(() => {
+    actions.fork(main);
```

Comment:
This is really great! For example to test the proxy server, it could be useful to not start main, but rather two servers and test the injection under various conditions (e.g. gzipped response). I think for a lot of tests we'll want to spin up the whole system, so having an easy way to do this is great, but this also makes it really easy to do more specific things if we need to!